Spigot-Server-Patches/0619-Added-SoundBuilder.patch

```diff
@@ -0,0 +1,307 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Fruxz <cedricspitzer@outlook.de>
+Date: Wed, 30 Dec 2020 02:23:06 +0100
+Subject: [PATCH] Added SoundBuilder
+
+
+diff --git a/src/main/java/io/papermc/paper/sound/PaperSoundBuilder.java b/src/main/java/io/papermc/paper/sound/PaperSoundBuilder.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..4ca7845d55e3e45420d941193e30b2ac9df7e5c8
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/sound/PaperSoundBuilder.java
+@@ -0,0 +1,234 @@
++package io.papermc.paper.sound;
++
++import com.google.common.collect.Lists;
++import net.minecraft.server.BaseBlockPosition;
++import net.minecraft.server.MinecraftKey;
++import net.minecraft.server.PacketPlayOutCustomSoundEffect;
++import net.minecraft.server.Vec3D;
++import org.bukkit.Location;
++import org.bukkit.Sound;
++import org.bukkit.SoundCategory;
++import org.bukkit.World;
++import org.bukkit.craftbukkit.entity.CraftPlayer;
++import org.bukkit.entity.Player;
++import org.bukkit.util.NumberConversions;
++
++import java.util.ArrayList;
++import java.util.Collection;
++import java.util.List;
++
++public class PaperSoundBuilder implements SoundBuilder {
++
++    private Sound sound;
++    private SoundCategory category;
++    private Collection<Player> receivers;
++    private Location location;
++    private float volume = 1F;
++    private float pitch = 1F;
++
++    public PaperSoundBuilder(Sound sound) {
++        this.sound = sound;
++    }
++
++    public PaperSoundBuilder(Sound sound, Float volume) {
++        this.sound = sound;
++        this.volume = volume;
++    }
++
++    public PaperSoundBuilder(Sound sound, Float volume, Float pitch) {
++        this.sound = sound;
++        this.volume = volume;
++        this.pitch = pitch;
++    }
++
++    public PaperSoundBuilder(Sound sound, SoundCategory category) {
++        this.sound = sound;
++        this.category = category;
++    }
++
++    public PaperSoundBuilder(Sound sound, SoundCategory category, Float volume) {
++        this.sound = sound;
++        this.category = category;
++        this.volume = volume;
++    }
++
++    public PaperSoundBuilder(Sound sound, SoundCategory category, Float volume, Float pitch) {
++        this.sound = sound;
++        this.category = category;
++        this.volume = volume;
++        this.pitch = pitch;
++    }
++
++    @Override
++    public PaperSoundBuilder play() {
++        if (this.location == null) {
```

Comment:
Make the location a required argument instead.

Reply:
But then, unfortunately, it loses a bit of its "builder"-ness

Reply:
You already require the sound. A location would be fine imho

Reply:
location is now required